pyphare/pyphare_tests/test_pharesee/test_geometry.py

```diff
@@ -0,0 +1,528 @@
+                          {"box": Box(-5, 5),"offset":(0,-65)},
+                        # right side overlap with periodicity, for all quantities
+                        {"box": Box(60, 70),"offset": (65,0)},
+                        {"box": Box(-5, 4),"offset":(0,-65)},
+                        {"box": Box(60, 69),"offset": (65,0 )},
+                        {"box": Box(-5, 4),"offset":(0,-65)},
+                        {"box": Box(60, 69),"offset": (65,0)},
+                        {"box": Box(-5, 4),"offset":(0,-65)},
+                        {"box": Box(60, 69),"offset":(65, 0)},
+                        {"box": Box(-5, 5),"offset":(0,-65)},
+                        {"box": Box(60, 70),"offset":(65, 0)},
+                        {"box": Box(-5, 5),"offset":(0,-65)},
+                        {"box": Box(60, 70),"offset":(65, 0)},
+                        {"box": Box(-1,0),"offset":(0,-65)},
+                        {"box": Box(64, 65),"offset":(65 ,0)}
+                    ],
+
+            1:  # level 1
+                [
+                    {"box": Box(59, 65),'offset': (0, 0)},
+                    {"box": Box(59, 64),"offset": (0, 0)},
+                    {"box": Box(59, 64),"offset": (0, 0)},
+                    {"box": Box(59, 64),'offset': (0, 0)},
+                    {"box": Box(59, 65),"offset": (0, 0)},
+                    {"box": Box(59, 65),"offset": (0, 0)}
+                ]
+
+
+        }
+
+
+        overlaps = hierarchy_overlaps(self.hierarchy)
+
+        for ilvl, lvl in enumerate(self.hierarchy.patch_levels):
+            self.assertEqual(len(expected[ilvl]), len(overlaps[ilvl]))
+
+            for exp, actual in zip(expected[ilvl], overlaps[ilvl]):
+
+                act_box    = actual["box"]
+                act_offset = actual["offset"]
+                exp_box    = exp["box"]
+                exp_offset = exp["offset"]
+
+                self.assertEqual(act_box, exp_box)
+                self.assertEqual(act_offset, exp_offset)
+
+
+
+
+    def test_touch_border(self):
+
+        self.assertFalse(touch_domain_border(Box(10,20), self.hierarchy.domain_box, "upper"))
+        self.assertFalse(touch_domain_border(Box(10, 20), self.hierarchy.domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box(0, 20), self.hierarchy.domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box(-5, 20), self.hierarchy.domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box(-5, 70), self.hierarchy.domain_box, "lower"))
+        self.assertTrue(touch_domain_border(Box(-5, 70), self.hierarchy.domain_box, "upper"))
+        self.assertTrue(touch_domain_border(Box(40, 70), self.hierarchy.domain_box, "upper"))
+        self.assertTrue(touch_domain_border(Box(40, 64), self.hierarchy.domain_box, "upper"))
+
+
+
+
+    def test_particle_ghost_area_boxes(self):
+
+        expected = {
+
+            0: [ {"boxes":[Box(33, 33), Box(-1,-1)]},
```

Comment:
what are these boxes size ? lower = upper ?

Reply:
the domain has 65 cells, so from 0 to 64. can you tell what's the "touch the border" rule, based on the examples ?